### PR TITLE
[GL] Improve WebGL version and feature checking

### PR DESCRIPTION
**Pull request by Gordon-F**
Fixes #issue
PR checklist:
- [ ] `make` succeeds (on *nix)
- [ ] `make reftests` succeeds
- [x] tested examples with the following backends: OpenGL/WebGL

### CODE REVIEW

**Comment by kvark** [comment]
Is there a particular reason we want to treat WebGL as a distinct target for the purposes of the compatibility checks?

**Comment by Gordon-F** [comment]
`WebGL` not always equals `OpenGL ES`. These changes should help properly check WebGL combability info. If we don't have `WebGL` as a distinct target we should set current `Version::new_embedded` (3.0/2.0) as a `WebGL` (1.0/2.0) somehow.

Also this changes can be a part of https://github.com/gfx-rs/wgpu/issues/1244.

**Comment by kvark** [comment]
> WebGL not always equals OpenGL ES

I know this is true, but the question is whether these differences are important for our GL backend. Could you give an example of where this distinction would make sense?

> If we don't have WebGL as a distinct target we should set current Version::new_embedded (3.0/2.0) as a WebGL (1.0/2.0) somehow.

Right, I don't think this is going to be too hard.

**Comment by Gordon-F** [comment]
> I know this is true, but the question is whether these differences are important for our GL backend. Could you give an example of where this distinction would make sense?

Ok, Agree with you. They are only theoretically different.

If other code in PR is ok, my plan:
1. Remove `new_webgl` from Version
2. Make `WebGL1` == `OpenGL ES 2.0` and `WebGL2` == `OpenGL ES 3.0` internally

**Comment by Gordon-F** [comment]
Temporarily convert PR to draft because latest changes (`WebGL 2.0` == `OpenGL ES 3.0`) panic in `cube` example.

**Comment by Gordon-F** [comment]
Code panic here: https://github.com/gfx-rs/gfx/blob/4b63f30807204b22f1ba52bb3356e69e19fb896a/src/backend/gl/src/lib.rs#L552-L559

I don't know how we can enable `FRAMEBUFFER_SRGB` for WebGL target 🤔

**Comment by Gordon-F** [comment]
Ok, we don't need to enable `FRAMEBUFFER_SRGB` on embedded targets. Should be ready to merge.

File: src/backend/gl/src/info.rs [hunk 1]
```diff
@@ -55,28 +55,43 @@ impl Version {
...
-        // TODO: Parse version and optional vendor
         let webgl_sig = "WebGL ";
-        let is_webgl = src.contains(webgl_sig);
+        // According to the WebGL specification
+        // VERSION	WebGL<space>1.0<space><vendor-specific information>
+        // SHADING_LANGUAGE_VERSION	WebGL<space>GLSL<space>ES<space>1.0<space><vendor-specific information>
+        let is_webgl = src.starts_with(webgl_sig);
+        let is_es;
+
         if is_webgl {
```
Comment (kvark) [inline src/backend/gl/src/info.rs:68]: nit: would be cleaner to do `let is_es = if is_webgl {... } else {..};`

File: src/backend/gl/src/queue.rs [hunk 2]
```diff
@@ -987,7 +987,7 @@ impl Queue {
                     _ => unsafe { gl.disable(gl_offset) },
                 }
 
-                if !self.share.info.is_webgl() && !self.share.info.version.is_embedded {
+                if !crate::is_webgl() && !self.share.info.version.is_embedded {
```
Comment (kvark) [inline src/backend/gl/src/queue.rs:990]: looks like `is_webgl` here is redundant. If it's not ES, then it's not WebGL for sure

File: src/backend/gl/src/lib.rs [hunk 3]
```diff
@@ -671,3 +672,7 @@ fn resolve_sub_range(
     let end = sub.size.map_or(whole.end, |s| whole.start + sub.offset + s);
     whole.start + sub.offset..end
 }
+
+pub(crate) const fn is_webgl() -> bool {
```
Comment (kvark) [inline src/backend/gl/src/lib.rs:676]: no need for `pub(crate)` here. Private functions are automatically visible to child modules

File: src/backend/gl/src/lib.rs [hunk 4]
```diff
@@ -553,6 +553,7 @@ impl adapter::PhysicalDevice<Backend> for PhysicalDevice {
             .0
             .legacy_features
             .contains(info::LegacyFeatures::SRGB_COLOR)
```
Comment (kvark) [inline src/backend/gl/src/lib.rs:555]: can the `LevacyFeatures::SRGB_COLOR` bit already take into account that it's not available on ES? seems strange that we have this `is_embedded` check in addition to it.
Also, something tells me that ES really does support SRGB. Maybe not all ES?

File: src/backend/gl/src/info.rs [hunk 5]
```diff
@@ -55,28 +55,43 @@ impl Version {
...
+                }
+                None => false,
+            };
         }
 
-        let es_sig = " ES ";
-        let is_es = match src.rfind(es_sig) {
+        let glsl_es_sig = "GLSL ES ";
+        let is_glsl;
+        match src.find(glsl_es_sig) {
```
Comment (kvark) [inline src/backend/gl/src/info.rs:85]: same here, let's do it more functional style: `let is_glsl = if {... } else {..};`

**Review by kvark — COMMENTED** [review]
Looks great overall, thank you for updating!
There is a few small things we can revisit before landing.

File: src/backend/gl/src/lib.rs (hunk 4 above)
Comment (Gordon-F) [reply src/backend/gl/src/lib.rs:555]: Based on info in [this extension](https://www.khronos.org/registry/OpenGL/extensions/EXT/EXT_sRGB_write_control.txt), `FRAMEBUFFER_SRGB` is always enabled with `OpenGL ES 3.0`. Added extra comment to code.

**Pull request merged**